File: src/main/java/org/jukeboxmc/block/BlockTripwire.java
package org.jukeboxmc.block;

import org.jukeboxmc.item.ItemTripwire;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class BlockTripwire extends BlockWaterlogable {

    public BlockTripwire() {
        super( "minecraft:tripwire_hook" );
    }

    @Override
    public ItemTripwire toItem() {
        return new ItemTripwire();
    }

    @Override
    public BlockType getBlockType() {
        return BlockType.TRIPWIRE;
    }

    @Override
    public boolean isSolid() {
        return false;
    }

    @Override
    public boolean isTransparent() {
        return true;
    }

    public void setPowered( boolean value ) {
        this.setState( "powered_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isPowered() {
        return this.stateExists( "powered_bit" ) && this.getByteState( "powered_bit" ) == 1;
    }

    public void setSuspended( boolean value ) {
        this.setState( "suspended_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isSuspended() {
        return this.stateExists( "suspended_bit" ) && this.getByteState( "suspended_bit" ) == 1;
    }

    public void setDisarmed( boolean value ) {
        this.setState( "disarmed_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isDisarmed() {
        return this.stateExists( "disarmed_bit" ) && this.getByteState( "disarmed_bit" ) == 1;
    }

    public void setAttached( boolean value ) {
        this.setState( "attached_bit", value ? (byte) 1 : (byte) 0 );
    }

    public boolean isAttached() {
        return this.stateExists( "attached_bit" ) && this.getByteState( "attached_bit" ) == 1;
    }
}

File: JukeboxMC-Server/src/main/kotlin/org/jukeboxmc/server/block/behavior/BlockTripWire.kt
package org.jukeboxmc.server.block.behavior

import org.cloudburstmc.nbt.NbtMap
import org.jukeboxmc.api.Identifier
import org.jukeboxmc.api.block.TripWire
import org.jukeboxmc.server.block.JukeboxBlock
import org.jukeboxmc.server.extensions.toByte

class BlockTripWire(identifier: Identifier, blockStates: NbtMap?) : JukeboxBlock(identifier, blockStates), TripWire {

    override fun canPassThrough(): Boolean {
        return true
    }

   override fun isAttached(): Boolean {
       return this.getBooleanState("attached_bit")
   }

   override fun setAttached(value: Boolean): TripWire {
       return this.setState("attached_bit", value.toByte())
   }

   override fun isPowered(): Boolean {
       return this.getBooleanState("powered_bit")
   }

   override fun setPowered(value: Boolean): TripWire {
       return this.setState("powered_bit", value.toByte())
   }

   override fun isSuspended(): Boolean {
       return this.getBooleanState("suspended_bit")
   }

   override fun setSuspended(value: Boolean): TripWire {
       return this.setState("suspended_bit", value.toByte())
   }

   override fun isDisarmed(): Boolean {
       return this.getBooleanState("disarmed_bit")
   }

   override fun setDisarmed(value: Boolean): TripWire {
       return this.setState("disarmed_bit", value.toByte())
   }
}